test/e2e/clusternetworkpolicy_test.go

```diff
@@ -0,0 +1,558 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	secv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/security/v1alpha1"
+	. "github.com/vmware-tanzu/antrea/test/e2e/utils"
+)
+
+// common for all tests.
+var (
+	allPods          []Pod
+	k8sUtils         *KubernetesUtils
+	allTestList      []*TestCase
+	pods, namespaces []string
+	podIPs           map[string]string
+	p80, p81         int
+)
+
+const (
+	// provide enough time for policies to be enforced & deleted by the CNI plugin.
+	networkPolicyDelay = 2 * time.Second
+)
+
+func failOnError(err error, t *testing.T) {
+	if err != nil {
+		log.Errorf("%+v", err)
+		k8sUtils.Cleanup(namespaces)
+		t.Fatalf("test failed: %v", err)
+	}
+}
+
+// TestCase is a collection of TestSteps to be tested against.
+type TestCase struct {
+	Name  string
+	Steps []*TestStep
+}
+
+// TestStep is a single unit of testing spec. It includes the CNP specs that need to be
+// applied for this test, the port to test traffic on and the expected Reachability matrix.
+type TestStep struct {
+	Name         string
+	Reachability *Reachability
+	CNPs         []*secv1alpha1.ClusterNetworkPolicy
+	Port         int
+	Duration     time.Duration
+}
+
+func initialize(t *testing.T, data *TestData) {
+	p80 = 80
+	p81 = 81
+	pods = []string{"a", "b", "c"}
+	namespaces = []string{"x", "y", "z"}
+
+	for _, podName := range pods {
+		for _, ns := range namespaces {
+			allPods = append(allPods, NewPod(ns, podName))
+		}
+	}
+	err := enableCNP(data)
+	failOnError(err, t)
+	k8sUtils, err = NewKubernetesUtils(data)
+	failOnError(err, t)
+	ips, err := k8sUtils.Bootstrap(namespaces, pods)
+	failOnError(err, t)
+	podIPs = *ips
+}
+
+func enableCNP(data *TestData) error {
```

Comment:
can you add a TODO comment with a link to https://github.com/vmware-tanzu/antrea/issues/893?

Reply:
Done.